frontends/mit-learn/src/pages/ChannelPage/ChannelPage.test.tsx

```diff
@@ -262,27 +312,59 @@ describe.each(NON_UNIT_CHANNEL_TYPES)(
           { level: 3, name: "Search Results" },
         ])
       })
-    })
+    }, 10000)
   },
 )
 
 describe("Channel Pages, Topic only", () => {
   test("Subtopics display", async () => {
-    const { channel, subTopics } = setupApis({
+    const { channel } = setupApis({
       search_filter: "topic=Physics",
       channel_type: ChannelTypeEnum.Topic,
     })
-    renderTestApp({ url: `/c/${channel.channel_type}/${channel.name}` })
+    const { topic, subTopics } = setupTopicApis(channel)
+    invariant(topic)
+    renderTestApp({
+      url: `/c/${channel.channel_type}/${channel.name.replace(/\s/g, "-")}`,
+    })
 
-    invariant(subTopics)
+    const subTopicsTitle = await screen.findByText("Subtopics")
+    expect(subTopicsTitle).toBeInTheDocument()
     const links = await screen.findAllByRole("link", {
       // name arg can be string, regex, or function
       name: (name) => subTopics?.results.map((t) => t.name).includes(name),
     })
-    links.forEach((link, i) => {
+    links.forEach(async (link, i) => {
       expect(link).toHaveAttribute("href", subTopics.results[i].channel_url)
     })
-  })
+  }, 10000)
+
+  test("Related topics display", async () => {
+    const { channel } = setupApis({
+      search_filter: "topic=Physics",
+      channel_type: ChannelTypeEnum.Topic,
+    })
+    const { subTopics, subTopicChannels } = setupTopicApis(channel)
+    invariant(subTopicChannels)
+    const subTopicChannel = subTopicChannels[0]
+    const filteredSubTopics = subTopics?.results.filter(
+      (t) =>
+        t.name.replace(/\s/g, "-") !== subTopicChannel.name.replace(/\s/g, "-"),
```

Comment:
Could be `t.name !== subtopicChannel.title` ... topic name and channel title are equivalent. (Topic model should use title, I guess. Oh well).